test/test/jfr/JfrMutliModeProfiling.java

```diff
@@ -0,0 +1,55 @@
+package test.jfr;
+
+import java.util.Random;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+/**
+ * Process to simulate lock contention.
+ */
+public class JfrMutliModeProfiling {
+    static Object sink;
+    static volatile int count = 0;
+
+    public static void main(String[] args) throws InterruptedException {
+        ExecutorService executor = Executors.newFixedThreadPool(3);
+        IntStream.range(0, 100_000).forEach(i -> executor.submit(JfrMutliModeProfiling::increment));
+        stop(executor);
+        allocate();
+    }
+
+    static synchronized void increment() {
+        count += 1;
+    }
+
+    static void stop(ExecutorService executor) {
+        try {
+            executor.shutdown();
+            executor.awaitTermination(10, TimeUnit.SECONDS);
+        }
+        catch (InterruptedException e) {
```

Comment:
You may remove all try-catch stuff for brevity. `TestRunner` already handles potential timeouts, failing the test in this case rather than ignoring exceptions.